File: 929.js
/**
 * @param {string[]} emails
 * @return {number}
 */
var numUniqueEmails = function(emails) {
    var map = {};
    for (var i = 0; i < emails.length; i++) {
        var email = emails[i].split("@");
        var local = email[0];
        local = local.split("+")[0];
        local = local.replace(new RegExp("\\.", "g"),"");
        var domain = email[1];
        map[local+domain] = true;
    }
    return Object.keys(map).length;
};

File: 929.java
import java.util.HashSet;
import java.util.Set;

class Solution {
  public int numUniqueEmails(String[] emails) {
    Set<String> res = new HashSet<>();
    for (String e : emails) {
      String[] s = e.split("@");
      String l = s[0].split("\\+")[0].replaceAll("\\.", "");
      res.add(l + "@" + s[1]);
    }
    return res.size();
  }
}